src/common/src/util/value_encoding/column_aware_row_encoding.rs

```diff
@@ -268,3 +287,89 @@ impl ValueRowDeserializer for ColumnAwareSerde {
         self.deserializer.deserialize(encoded_bytes)
     }
 }
+
+/// Deserializes row `encoded_bytes`, drops columns not in `valid_column_ids`, serializes and returns.
```

Comment:
IIUC, we don't do "real" deserialization here. We're only working with the serialized bytes and drop the unnecessary ones then rearrange them.